### PR TITLE
Lab 6 

### PR DESCRIPTION
**Pull request by SoulHowl**
Barsukevich S. 953504. Lab6.  Created SPA with React.
link https://weather-8b64a.web.app/

### CODE REVIEW

File: lab6/react-spa/src/components/auth/c_auth.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React, {useRef} from 'react';
+import { login, signUp} from "../all/api/firebase";
+import {useAuth} from "../all/api/useAuth";
+import User from "../user/c_user";
+
+function Auth() {
+    const {isSignedIn} = useAuth()
+    const email = useRef("dsf");
```
Comment (katetushkan) [inline lab6/react-spa/src/components/auth/c_auth.jsx:8]: why r u using refs here?

File: lab6/react-spa/src/components/forecast/c_forecast.jsx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+                        modalFooterTemplate = '<div class="modal__footer">{{buttons}}</div>',
+                        modalButtonTemplate = '<button type="button" class="{{button_class}}" data-handler={{button_handler}}>{{button_text}}</button>',
+                        modalHTML,
+                        modalFooterHTML = '';
+
+                    elemModal.classList.add('modal');
+                    elemModal.id = "fuck";
+                    let rose = '<svg xmlns="http://www.w3.org/2000/svg" width="600" height="600" viewBox="0 0 600 600"><rect width="100%" height="100%" fill="#fff" /></svg>';//document.getElementById("fake").innerHTML
+                    modalHTML = modalTemplate.replace('{{title}}', options.title || 'Simple Wind rose, according to latest data');
+                    modalHTML = modalHTML.replace('{{content}}', options.content || rose);
```
Comment (katetushkan) [inline lab6/react-spa/src/components/forecast/c_forecast.jsx]: u need to make a component called `Dialog` and then make a conditional render

File: lab6/react-spa/src/components/main/main.jsx [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+//###########################################################
+    function update_weather(current)
+    {
+        update_widget_temp(current.temperature,current.state)
+        update_widget_humidity(current.humidity)
+        update_pressure_widget(current.pressure)
+        update_wind_widget(current.wind_str,current.wind_dir)
+        change_wallPaper(current.state)
+    }
+
```
Comment (katetushkan) [inline lab6/react-spa/src/components/main/main.jsx]: all ur widgets better be a components...

File: lab6/react-spa/src/components/main/main.jsx [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+                    <section className="find">
+                        <section className="top-banner">
+                            <div className="container-find">
+                                <form className="find-form">
+                                    <input  type="text" placeholder="Search for a city" autoFocus/>
+                                    <button className="button-find" type="submit">SUBMIT</button>
+                                    <span className="msg"></span>
+                                </form>
+                            </div>
+                        </section>
```
Comment (katetushkan) [inline lab6/react-spa/src/components/main/main.jsx]: Form should be a component.

**Review by katetushkan — CHANGES_REQUESTED** [review]
I think u need to read [this](https://overreacted.io/writing-resilient-components/)

File: lab6/react-spa/src/components/auth/c_auth.jsx (hunk 1 above)
Comment (SoulHowl) [reply lab6/react-spa/src/components/auth/c_auth.jsx:8]: removed refs

File: lab6/react-spa/src/components/forecast/c_forecast.jsx (hunk 2 above)
Comment (SoulHowl) [reply lab6/react-spa/src/components/forecast/c_forecast.jsx]: created component c_dialog

File: lab6/react-spa/src/components/main/main.jsx (hunk 3 above)
Comment (SoulHowl) [reply lab6/react-spa/src/components/main/main.jsx]: created component for each widget

File: lab6/react-spa/src/components/main/main.jsx (hunk 4 above)
Comment (SoulHowl) [reply lab6/react-spa/src/components/main/main.jsx]: also created component for this form

**Review by katetushkan — APPROVED** [review]
Okay, <kbd>8</kbd> for u, Ur average is <kbd>8.5</kbd>.

**Review by katetushkan — APPROVED** [review]
.

**Comment by SoulHowl** [comment]
> Okay, 8 for u, Ur average is 8.5.

Okay (ง ͡° ͜ʖ ͡°)ง

**Pull request merged**